enumerable.rb

```diff
@@ -0,0 +1,163 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  # 1 my_each
+  def my_each
+    to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+  end
```

Comment:
#my_each returns an Enumerator if no block is given
#my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
#my_each when a block is given when self is a range returns the range itself after calling the 
given block once for each element in self